hips/hip-9999.md

```diff
@@ -0,0 +1,80 @@
+---
+hip: 9999
+title: "Helm Triage Maintainers"
+authors: [ "Matthew Fisher <matt.fisher@microsoft.com>" ]
+created: "2021-07-06"
+type: "process"
+status: "draft"
+---
+
+## Abstract
+
+The existing Helm Project Maintainers have identified the need for a new "triage" role. This document lists out the responsibilities of this new role.
+
+## Motivation
+
+Helm Project Maintainers are responsible for activities surrounding the development and release of code, the operation of any services they own (e.g. <https://get.helm.sh>), or the tasks needed to execute their project (e.g., community management, setting up an event booth). Introducing a new "Helm Triage Maintainer" role allows new project maintainers to join the project without the responsibilities of release management, service operation, or event management.
+
+This new role provides three benefits:
+
+1. New triage maintainers can help maintain the project's day-to-day, such as responding to new tickets and review pull requests
+2. Helm project maintainers' burden is reduced, allowing them to carry out higher-level tasks like coordinating software releases, organizing events, and maintaining production services.
+3. Provides an easier on-ramp from community members, to triage maintainers, to project maintainers, to org maintainers.
+
+## Rationale
+
+Project maintainers are looking for new members to join the project. However, community members have expressed in the past that they cannot commit more than a few hours per month to the project. Additionally, some project maintainers are concerned providing newer members the "keys to the castle" prior to a formal vetting process may introduce significant risk to the project's users. A new maintainer may be available to help maintain the project, but it may take several months before trust is established.
+
+## Specification
+
+The new triage maintainer role is allowed to perform certain day-to-day tasks as a maintainer of the project, including:
+
+- attach labels to issues
+- assign issues to milestones
+- receive access to the cncf-helm-core-maintainers mailing list
+- vote on governance-related topics
+- review and approve pull requests
+- merge pull requests
```

Comment:
Is this more "project maintainer" status than triaging?

Reply:
The idea here is to offload most of the trivial day-to-day work. To me, merging pull requests into the release branch would fall under that category.

Triagers can help with reviewing and merging pull requests into the release branch. They just can't cut new releases.

Reply:
See also https://github.com/helm/community/pull/199#discussion_r665422505

Reply:
I am not sure that merging PRs is "trivial day-to-day work". I think other projects if using stepping stone approach to full maintainer, don't allow PR merging straight away.

Reply:
Let me clarify my position: triage maintainers should be able to do anything regular core maintainers can do UNLESS it requires protected information, access to secret data, or the actions of the triage maintainer could affect a user's environment.

Based on that argument, a triage maintainer SHOULD be allowed to merge a pull request and cherry-pick it into the release branch. If they are uncomfortable merging PRs, they can ask for help from another maintainer.

I do not want to be in the business of gate-keeping other maintainers from performing their best work because it's deemed "too hard". If the work is deemed "too hard", then it's either a good learning opportunity or an opportunity for the core maintainers to improve that workflow.

On the aspect of security: the job of the Release Maintainer is to [check and review each commit](https://helm.sh/docs/community/release_checklist/#3-majorminor-releases-commit-and-push-the-release-branch) in the release branch before being released to the public.

> If anyone is available, let others peer-review the branch before continuing to ensure that all the proper changes have been made and all of the commits for the release are there.

And because we track issues/PRs [through github milestones](https://github.com/helm/helm/milestone/110?closed=1), it should be easy for a release maintainer to check and see if extra commits "snuck in" or the release branch was tampered with.

Triage maintainers cannot cut new releases, so they cannot merge their own PRs and cut a new release. Therefore merging PRs is a safe work item for them to take on if they wish.

Does that help change your position on this?

Reply:
Sigh... I thought branch protection rules could be used to prevent maintainers from cutting new tags. Looks like that is not the case:

https://github.com/isaacs/github/issues/1091

Until that is available, it looks like we may need to revoke write access for triage maintainers. Otherwise they will be able to cut tags and affect production.

Reply:
> Let me clarify my position: triage maintainers should be able to do anything regular core maintainers can do UNLESS it requires protected information, access to secret data, or the actions of the triage maintainer could affect a user's environment.

Ok, thanks for the clarification @bacongobbler. I understand the role better now.